docs/source/en/tasks/prompting.md

```diff
@@ -0,0 +1,466 @@
+
+You also need to allow remote code execution. This is because the Falcon models use a new architecture that is not part of transformers yet - instead, the code necessary is provided by the model authors in the repo.
+
+</Tip>
+
+Now that we have the model loaded via the pipeline, let's explore how you can use prompts to solve NLP tasks.
+
+#### Text classification
+
+One of the most common forms of text classification is sentiment analysis, which assigns a label like "positive", "negative", 
+or "neutral" to a sequence of text. Let's write a prompt that instructs the model to classify a given text (a movie review). 
+We'll start by giving the instruction, and then specifying the text to classify. Note that instead of leaving it at that, we're 
+also adding the beginning of the response - `"Sentiment: "`:
+
+```python
+>>> torch.manual_seed(0) # doctest: +IGNORE_RESULT
+>>> prompt = """Classify the text into neutral, negative or positive. 
+... Text: This movie is definitely one of my favorite movies of its kind. The interaction between respectable and morally strong characters is an ode to chivalry and the honor code amongst thieves and policemen.
+... Sentiment:
+... """
+
+>>> sequences = pipe(
+...     prompt,
+...     max_length=100,
+...     do_sample=True,
+...     top_k=10,
+...     num_return_sequences=1,
+...     eos_token_id=tokenizer.eos_token_id,
+... )
+
+>>> for seq in sequences:
+...     print(f"Result: {seq['generated_text']}")
+Result: Classify the text into neutral, negative or positive. 
+Text: This movie is definitely one of my favorite movies of its kind. The interaction between respectable and morally strong characters is an ode to chivalry and the honor code amongst thieves and policemen.
+Sentiment:
+Positive
+```
+
+As a result, the output contains a classification label from the list we have provided in the instructions, and it is a correct one!
+
+<Tip>
+
+You may notice that in addition to the prompt, we pass a number of text generation parameters. We cover these parameters 
+in other docs, such as the [Text generation strategies](../generation_strategies) guide. For the purposes of this guide, 
+you only need to know that  `max_length` controls the length of the total output (including the initial prompt). 
+</Tip>
+
+#### Named Entity Recognition
+
+Named Entity Recognition (NER) is a task of finding named entities in a piece of text, such as a person, location, or organization.
+Let's modify the instructions in the prompt to make the LLM perform this task. Here, let's also set `return_full_text = False` 
+so that output doesn't contain the prompt:
+
+```python
+>>> torch.manual_seed(1) # doctest: +IGNORE_RESULT
+>>> prompt = """Return a list of named entities in the text.
+... Text: The Golden State Warriors are an American professional basketball team based in San Francisco.
+... Named entities:
+... """
+
+>>> sequences = pipe(
+...     prompt,
+...     max_length=43,
+...     do_sample=True,
+...     top_k=10,
```

Comment:
```suggestion
```
Think for NER we also don't need to use sampling and should just advertise to use greedy_search